src/java.desktop/share/classes/javax/swing/TablePrintable.java

```diff
@@ -410,7 +413,7 @@ public int print(Graphics graphics, PageFormat pageFormat, int pageIndex)
             rMax = table.getRowCount();
         }
         int rowHeight = 0;
-        for(int visrow = rMin; visrow < rMax; visrow++) {
+        for(int visrow = rMin; visrow <= rMax; visrow++) {
```

Comment:
So was the issue here that this is a 1 based index not a zero-based index  and we basically had an off-by-one bug ?

nit : for( -> "for ("

Reply:
This issue is not related to the current bugfix though, it is related to the border rectangle drawn over the table at printing. Since I had corrected the maxRow computation in TablePrintable class, an existing issue where rectangle border drawing was getting affected, so in had to fix the issue. Since Rectangle border had to be drawn around the table rows the computations had to be from minRow till maxRow.